apps/challenges/views.py

```diff
@@ -340,3 +358,221 @@ def challenge_phase_split_list(request, challenge_pk):
+        BASE_LOCATION = tempfile.mkdtemp()
+        unique_folder_name = uuid.uuid4()
+        CHALLENGE_ZIP_DOWNLOAD_LOCATION = os.path.join('{}/{}.zip'.format(BASE_LOCATION, unique_folder_name))
+        if response and response.status_code == 200:
+            with open(CHALLENGE_ZIP_DOWNLOAD_LOCATION, 'w') as zip_file:
+                zip_file.write(response.content)
+
+    except:
+        response_data = {
+            'error': 'Some error occured during file upload. Please upload it again.'
+            }
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    # Extract zip file
+    try:
+        zip_ref = zipfile.ZipFile(CHALLENGE_ZIP_DOWNLOAD_LOCATION, 'r')
+        zip_ref.extractall('{}/{}'.format(BASE_LOCATION, unique_folder_name))
+        zip_ref.close()
+    except:
+        response_data = {
+            'error': 'The zip file contents cannot be extracted. Please check the format.'
+            }
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    # Search for yaml file
+    try:
+        for name in zip_ref.namelist():
+            if name.endswith('.yaml') or name.endswith('.yml'):
+                yaml_file = name
+                extracted_folder_name = yaml_file.split(os.path.basename(yaml_file))[0]
+    except:
+        response_data = {
+            'error': 'There is no yaml file in zip folder. Please add a yaml file.'
+        }
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    try:
+        with open('{}/{}/{}'.format(BASE_LOCATION, unique_folder_name, yaml_file), "r") as stream:
+            yaml_file_data = yaml.load(stream)
+    except yaml.YAMLError as exc:
+        response_data = {
+            'error': exc
+            }
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    # Check for evaluation script path in yaml file.
+    evaluation_script = yaml_file_data['evaluation_script']
+    if evaluation_script:
+        evaluation_script_path = os.path.join("{}/{}/{}{}".format(BASE_LOCATION,
+                                                                  unique_folder_name,
+                                                                  extracted_folder_name,
+                                                                  evaluation_script))
+    else:
+        response_data = {
+            'error': 'There is no path for evaluation script. Please add a path for it.'
+            }
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    # Check for evaluation script file in extracted zip folder.
+    if os.path.isfile(evaluation_script_path):
+        challenge_evaluation_script = open(evaluation_script_path, 'rb')
+        challenge_evaluation_script_file = File(challenge_evaluation_script)
+    else:
+        response_data = {
+            'error': 'There is no evaluation script file at the given path in yaml.'
```

Comment:
Can we update the message to `No evaluation script is present in the zip file. Please try uploading again the zip file after adding evaluation script`